### PR TITLE
Implement Event Scheduling

### PR DESCRIPTION
**Pull request by jessetilro**
Implemented fully functional event scheduling. Events are introduced with a distribution modelled as a poisson process. The queue of active events is monitored, and when an event expires it damages the ship and is discarded from the log. Finally, when the ship's health is depleted, the game is stopped.

Resolves issues #12 and #13.

It is not tested yet, so we have some work to do the upcoming sprint.

### CODE REVIEW

**Comment by dmalarev** [comment]
LGTM

File: src/main/java/pixelperfect/Game.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ public void simpleUpdate() {
    */
   @Override
   public void simpleUpdate(float tpf) {
```
Comment (FlorisXll) [inline src/main/java/pixelperfect/Game.java]: tpf means? (time per frame?)

File: src/main/java/pixelperfect/event/EventLog.java [hunk 2]
```diff
@@ -27,6 +27,7 @@ public EventLog() {
    */
   public void notify(Event event) {
     events.add(event);
+    System.out.println("The ship received a new event: " + event.getDescription());
```
Comment (dalderliesten) [inline src/main/java/pixelperfect/event/EventLog.java]: Looks good, I'd consider making this NEW EVENT for the next sprint, easier to view.

File: src/main/java/pixelperfect/Game.java (hunk 1 above)
Comment (jessetilro) [inline src/main/java/pixelperfect/Game.java]: Exactly. https://hub.jmonkeyengine.org/t/units-of-tpf/21172

**Comment by dalderliesten** [comment]
The test coverage is a bit lower, but we'll fix this in the coming sprint.  Approved.

File: src/main/java/pixelperfect/event/EventLog.java [hunk 3]
```diff
@@ -36,8 +47,26 @@ public void notify(Event event) {
...
+  public synchronized void discard(Event event) {
     events.remove(event);
   }
 
+  /**
+   * Evaluate the log to check whether events have expired since the last update.
+   */
+  public synchronized void update() {
+    ArrayList<Event> discardPile = new ArrayList<Event>();
+    for (Iterator<Event> it = events.iterator(); it.hasNext();) {
```
Comment (FlorisXll) [inline src/main/java/pixelperfect/event/EventLog.java]: Why not  for(Event it : events) {}.

Comment (dalderliesten) [inline src/main/java/pixelperfect/event/EventLog.java]: I would assume to prevent a concurrencyexception error.

**Pull request merged**